CMakeLists.txt

```diff
@@ -242,6 +243,9 @@ function(set_finufft_options target)
   endif()
   target_link_libraries(${target} PRIVATE xsimd)
   target_link_libraries(${target} PRIVATE ${FINUFFT_FFTLIBS})
+  if(FINUFFT_STATIC_LINKING AND FINUFFT_STATIC_LINKING_PIC)
+    set_property(TARGET ${target} PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
Close but this might break with `fftw`, `ducc`, dependencies should also be `-fPIC` :)

So we should use `set_property(TARGET ${target} PROPERTY POSITION_INDEPENDENT_CODE FINUFFT_STATIC_LINKING_PIC)` for all targets and this variable should be set to on when we do shared linking